image.md

```diff
@@ -0,0 +1,309 @@
+# Image
+
+This section defines a format for encoding a container's *filesytem bundle* as a portable *image*.
+An image is a set of files organized in a certain way, containing all of the necessary data and metadata, for any compliant runtime to verify the validity of the image's contents and to construct a filesystem bundle represented by the image.
+Images are meant to be portable such that they can be used to reliably move filesystem bundles between instances of OCI compliant implementations.
+
+In its simplest form, an image is a tar file of the filesystem bundle (root file system directory plus a configuration file) along with some additional metadata.
+While the runtime only sees a single directory as the root filesystem, it might actually be stored as a layering of filesytems under the covers.
+When representing this layering within an image, each "layer" will be independently referenced.
+
+## Image Contents
+
+A compliant image contains the following pieces of information:
+
+1. A `manifest.json` file describing the contents of the image.
+This file is REQUIRED to be present in the image.
+This file extends the [`config.json`](config.md) file with some additional properties:
```

Comment:
On Mon, Feb 15, 2016 at 06:40:49PM -0800, Doug Davis wrote:
> Or do we want to split these into multiple files.  I don't yet see
> value in splitting them.

The value I see in splitting files is clear separation of concerns.  A
manifest file contains only the bits needed for fetching and
validation.  A config.json contains only the bits needed by the
‘start’ command.  A process.json contains only the bits needed by the
‘exec’ command [1].  The state JSON contains only the bits needed to
manipulate (join, signal, tear down) a container.  I'm ok allowing
folks to wedge all this into one file [2], but it seems like it would
be easier to write the individual tools and to understand the results
of a particular action if each action had a separate config file.

Unless there's some overlap between the actions, I see no benefit to
configuring multiple actions in the same file.  And “start a container
from this OCI-bundle directory” and “create and OCI-bundle directory
from this tarball” don't seem like they have *any* overlap.

[1]: https://github.com/opencontainers/specs/pull/225/files#diff-b84a8d65d8ed53f4794cd2db7e8ea731R97
[2]: https://groups.google.com/a/opencontainers.org/d/msg/dev/p_u2xps6jvo/bcMI6hm1DwAJ
     Subject: Re: Labels and extension meta data in containers #108
     Date: Wed, 30 Dec 2015 20:54:37 -0800
     Message-ID: <20151231045437.GA6362@odin.tremily.us>